### PR TITLE
Refactor run mode tests [BA-5963]

### PR DESCRIPTION
**Pull request by aednichols**
We now invoke run mode from the command line instead of testing the actor in a special (and flakey) test harness.

### CODE REVIEW

**Comment by aednichols** [comment]
Adding @mcovarr to review since you expressed an interest yesterday. No rush and feel free to punt if you're busy.

File: src/ci/bin/testSingleWorkflowRunner.sh [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+source "${BASH_SOURCE%/*}/test.inc.sh" || source test.inc.sh
+
+cromwell::build::setup_common_environment
+
+cromwell::build::assemble_jars
+
+java -jar $CROMWELL_BUILD_CROMWELL_JAR run ./centaur/src/main/resources/standardTestCases/hello/hello.wdl --inputs ./centaur/src/main/resources/standardTestCases/hello/hello.inputs --metadata-output ./run_mode_metadata.json > console_output.txt
+
+# grep exits 1 if no matches
+grep "terminal state: WorkflowSucceededState" console_output.txt
```
[the remark this answers is not in the document]
Comment (cjllanwarne) [reply src/ci/bin/testSingleWorkflowRunner.sh]: Huh, I guess I must have - can confirm it *is* there when I look now

**Comment by cjllanwarne** [comment]
I'm satisfied as-is but I'll allow Miguel to supply the final 👍 once his comment on `HEREDOC`ing and `jq` is resolved

File: src/ci/bin/testSingleWorkflowRunner.sh [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+grep "\"wf_hello.hello.salutation\": \"Hello m'Lord!\"" console_output.txt
+
+grep "\"actualWorkflowLanguageVersion\": \"draft-2\"" run_mode_metadata.json
+grep "\"actualWorkflowLanguage\": \"WDL\"" run_mode_metadata.json
+grep "\"wf_hello.hello.salutation\": \"Hello m'Lord!\"" run_mode_metadata.json
+grep "\"wf_hello.hello.addressee\": \"m'Lord\"" run_mode_metadata.json
+grep "\"dockerImageUsed\": \"ubuntu@sha256:71cd81252a3563a03ad8daee81047b62ab5d892ebbfbf71cf53415f29c130950\"" run_mode_metadata.json
```
[the remark this answers is not in the document]
Comment (aednichols) [reply src/ci/bin/testSingleWorkflowRunner.sh]: Neat, TIL!

Heredoc looks super useful but weird name, no?

File: src/ci/bin/testSingleWorkflowRunner.sh [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+{
+  "actualWorkflowLanguage": "WDL",
+  "actualWorkflowLanguageVersion": "draft-2",
+  "dockerImageUsed": "ubuntu@sha256:71cd81252a3563a03ad8daee81047b62ab5d892ebbfbf71cf53415f29c130950",
+  "addressee": "m'Lord",
+  "salutation": "Hello m'Lord!"
+}
+FIN
+
+jq '{actualWorkflowLanguage,actualWorkflowLanguageVersion,dockerImageUsed:.calls["wf_hello.hello"][0].dockerImageUsed,addressee:.inputs["wf_hello.hello.addressee"],salutation:.outputs["wf_hello.hello.salutation"]}' run_mode_metadata.json > actual.json
```
Comment (mcovarr) [inline src/ci/bin/testSingleWorkflowRunner.sh]:
```suggestion
jq '{
  actualWorkflowLanguage,
  actualWorkflowLanguageVersion,
  dockerImageUsed:.calls["wf_hello.hello"][0].dockerImageUsed,
  addressee:.inputs["wf_hello.hello.addressee"],
  salutation:.outputs["wf_hello.hello.salutation"]
}' run_mode_metadata.json > actual.json
```

Comment (aednichols) [reply src/ci/bin/testSingleWorkflowRunner.sh]: Yes, that is a considerable improvement!

**Pull request merged**